rust/polars-ocaml-macros/src/lib.rs

```diff
@@ -4,6 +4,10 @@ use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
 use syn::{parse_macro_input, punctuated::Punctuated};
 
+// TODO: a common mistake when using the attribute macro is to specify OCaml<_>
+// for arguments or OCamlRef<_> for return types, which should never happen.
+// In these cases, the macro should probably point out this issue and suggest
+// what to do (use the other type).
 fn ocaml_interop_export_implementation(item_fn: syn::ItemFn) -> TokenStream2 {
```

Comment:
(pro tip) avoid panicking in proc macros as it will break rust-analyzer. Instead, you can emit a stub code (the better the stub reflects stuff inside the macro more) and `compile_error!`:
https://doc.rust-lang.org/std/macro.compile_error.html

Reply:
Yeah, I was lazy and didn't bother doing that, especially since I'm not as used to doing result-monad-like error propagation in Rust properly. Adding a TODO to address this in the future (I'm not convinced that it's worth putting a ton of effort getting the dev experience of using this macro perfect especially with so many other things on the stack!).